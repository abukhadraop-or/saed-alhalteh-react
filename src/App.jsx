import SignInForm from 'components/Forms/SignInForm';
import SignUpForm from 'components/Forms/SignUpForm';
import HomePage from 'components/HomePage/HomePage';
import NavBar from 'components/NavBar/NavBar';
import Post from 'components/Post/Post';
import UserProfile from 'components/UserProfile/UserProfile';
import React from 'react';
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import GlobalStyle from 'styles/global-style';
/**
 * The entry point for the application.
 *
 * @return {JSX.Element} App component.
 */
function App() {
  return (
    <BrowserRouter>
      <GlobalStyle />
      <NavBar />
      <Switch>
        <Route path="/user" component={UserProfile} />
        <Route path="/posts/" component={Post} />
        <Route path="/signUp" component={SignUpForm} />
        <Route path="/signIn" component={SignInForm} />
        <Route path="/" exact component={HomePage} />
      </Switch>
    </BrowserRouter>
  );
}

export default App;
