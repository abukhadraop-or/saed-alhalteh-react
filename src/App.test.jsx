import App from 'App';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';

describe('<App /> tests', () => {
  it('matches the snapshot', () => {
    const wrapper = shallow(<App />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
