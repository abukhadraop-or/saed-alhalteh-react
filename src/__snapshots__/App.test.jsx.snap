// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> tests matches the snapshot 1`] = `
<BrowserRouter>
  <Memo(l) />
  <NavBar />
  <Switch>
    <Route
      component={[Function]}
      path="/user"
    />
    <Route
      component={[Function]}
      path="/posts/"
    />
    <Route
      component={[Function]}
      path="/signUp"
    />
    <Route
      component={[Function]}
      path="/signIn"
    />
    <Route
      component={[Function]}
      path="/not-found"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Redirect
      to="/not-found"
    />
  </Switch>
</BrowserRouter>
`;
